pkg/cmd/status/status.go

```diff
@@ -286,23 +286,35 @@ func (s *StatusGetter) LoadNotifications() error {
 					if !ok {
 						return nil
 					}
-					if actual, err := s.ActualMention(n.Subject.LatestCommentURL); actual != "" && err == nil {
+					var preview string
+					if actual, err := s.ActualMention(n.Subject.LatestCommentURL); err == nil {
+						preview = actual
+					} else {
+						var httpErr api.HTTPError
+						if !errors.As(err, &httpErr) {
+							abortFetching()
+							return fmt.Errorf("could not fetch comment: %w", err)
+						}
+						if httpErr.StatusCode == 403 {
+							s.addAuthError(httpErr.Message, factory.SSOURL())
+							return nil
+						} else if httpErr.StatusCode == 404 {
+							preview = "(could not find comment)"
+						} else {
+							abortFetching()
+							return fmt.Errorf("could not fetch comment: %w", err)
+						}
+					}
+
+					if preview != "" {
```

Comment:
**Would you find this more readable?** 🙇

I agree with you that a `404` comment means the user cannot fetch it shouldn't stop the `gh status` output. 👍  Aside from a minor change in the wording &mdash; _(keeping the verbiage consistent and it isn't like we can't find it but can't fetch it)_ &mdash; I think this is more idiomatic handling the error immediately while also allowing us to handle http and non-http errors without duplicating the outcomes.

```suggestion
					actual, err := s.ActualMention(n.Subject.LatestCommentURL)

					if err != nil {
						var httpErr api.HTTPError
						httpStatusCode := -1

						if errors.As(err, &httpErr) {
							httpStatusCode = httpErr.StatusCode
						}

						switch httpStatusCode {
						case 403:
							s.addAuthError(httpErr.Message, factory.SSOURL())
							return nil
						case 404:
							actual = "(could not fetch comment)"
						default:
							abortFetching()
							return fmt.Errorf("could not fetch comment: %w", err)
						}
					}

					if actual != "" {
```

Reply:
Sure — your suggestion looks pretty equivalent to me. I also didn't like the duplicated `fmt.Errorf` in mine, but was also hesitant about a mutated + dummy status code var vs single assignment, and then decided not to spend more time on it. I'll add your changes.